sky/clouds/service_catalog/data_fetchers/fetch_aws.py

```diff
@@ -334,9 +341,11 @@ def fetch_availability_zone_mappings() -> pd.DataFrame:
     parser.add_argument('--no-az-mappings',
                         dest='az_mappings',
                         action='store_false')
-    parser.add_argument('--check-regions-integrity',
-                        action='store_true',
-                        help='Check the region integrity of the fetched data.')
+    parser.add_argument(
+        '--check-regions-match-request',
+        action='store_true',
+        help=('Check whether the fetched data contains all regions '
+              'requested.'))
```

Comment:
```suggestion
        help=('Check whether the fetched data contains all regions '
              'requested, to guard against network issues/glitches.'))
```

How about call it `--strict` or `--check-fetched-results`? IMO it seems a bit easier to parse.